blog/_posts/2017-02-22-repl-0.6-highlights.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: post
+title:  "Upgrades to the REPL in v0.6"
+author: <a href="https://kristofferc.github.io/">Kristoffer Carlsson</a>
+---
+
+Since v0.3, Julia comes with a REPL completely written in Julia ([PR #6270](https://github.com/JuliaLang/julia/pull/6270)).
+It contains, among other features, code completion, history with reverse search, a rich set of keybindings which in addition are customizable (see the [Interacting with Julia](http://docs.julialang.org/en/stable/manual/interacting-with-julia/) part of the manual for REPL documentation).
+Even so, v0.6 of Julia comes with quite a few new visual tweaks and feature upgrades over v0.5.
+In this blog post, we look into some of these changes and customizations to the Julia REPL available in v0.6.
+
+## Prompt pasting
+
+It is quite common (as in the julia manual and in subsequent code blocks of this blog post) to write snippets of julia code like they were copied straight from a REPL session, for example:
+
+```jl
+julia> f() = rand(10^6)
+f (generic function with 1 method)
+
+julia> @time f();
+  0.052948 seconds (25.05 k allocations: 8.769 MiB)
+```
+
+If you wanted to try out this code in v0.5, you would typically copy the code into a text editor, remove the `julia> ` prompt, remove the output, and then finally have code that looked like.
+
+```jl
+f() = rand(10^6);
+
+@time f();
+```
+
+which could then be pasted back into the REPL to be executed.
+In v0.6, the REPL comes with a feature called *prompt pasting* by which a pasted expression starting with `julia> ` automatically has prompts and eventual output scrubbed before being executed.
+If you have a v0.6 REPL currently at hands, you can try to copy and paste the example with the prompt and output above directly into the REPL and you will see that it works in exactly the same way as if we had manually removed them.
+
+Unfortunately, for technical reasons (not supporting bracket pasting), this feature does not work on the standard cmd-terminal in windows. There are however multiple alternative terminal emulators on Windows that do support this, e.g. [Console2](https://sourceforge.net/projects/console/), [Cygwin](https://www.cygwin.com/) and [Cmder](http://cmder.net/).
+
+## Printing of stacktraces
+
+The accuracy of the stacktraces getting printed when an error is thrown is improved in every release. Function calls are now even tracked through inlining so that often the full chain of calls up to the error is showed.
+This, in combination with the common way of writing Julia code by which one frequently write functions that does a small amount of job and then dispatch to another function makes for quite deep stacktraces.
```

Comment:
"This, combined with the Julia idiom of breaking complex tasks across many smaller functions that perform simpler tasks, makes for potentially deep stacktraces." ?